todoman/ui.py

```diff
@@ -200,22 +200,47 @@ class TodoFormatter:
         "{id:3d} [{completed}] {urgent} {due} {summary} {list}{percent}"
     # compact_format = "{completed} {urgent}  {due}  {summary}"
 
-    def __init__(self, date_format):
+    def __init__(self, date_format, time_format, dt_separator):
         self.date_format = date_format
+        self.time_format = time_format
+        self.dt_separator = dt_separator
+        self.datetime_format = dt_separator.join(filter(bool, (
+            date_format, time_format
+        )))
+
         self._localtimezone = tzlocal()
-        self.now = datetime.now().replace(tzinfo=self._localtimezone)
-        self.tomorrow = self.now.date() + timedelta(days=1)
+        self.now = datetime.datetime.now().replace(tzinfo=self._localtimezone)
 
         # An empty date which should be used in case no date is present
         self.date_width = len(self.now.strftime(date_format))
         self.empty_date = " " * self.date_width
-        # Map special dates to the special string we need to return
-        self.special_dates = {
-            self.now.date(): "Today".rjust(self.date_width, " "),
-            self.tomorrow: "Tomorrow".rjust(self.date_width, " "),
-        }
+
+        self.time_width = len(self.now.strftime(time_format))
+        self.empty_time = " " * self.time_width
+        self._prepare_special_datetimes([
+            (self.now.date(),
+             ("Today", self.time_format)),
+            (self.now.date() + datetime.timedelta(days=1),
+             ("Tomorrow", self.time_format)),
+            (self.now.date() - datetime.timedelta(days=1),
+             ("Yesterday", self.time_format)),
+            (None, (self.empty_date, self.empty_time)),
+        ])
+
         self._parsedatetime_calendar = parsedatetime.Calendar()
 
+    def _prepare_special_datetimes(self, mapping):
+        # Map special dates to the special string we need to return
+        self.special_datetimes = {}
+        for key, value in mapping:
+            if isinstance(value, tuple):
```

Comment:
Fixed. Left-over when I was trying to add `"in one hour"` (but as said that would be only implementable as `if-else`).